src/main/java/org/apache/commons/collections4/list/ExtendedTreeList.java

```diff
@@ -0,0 +1,284 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.collections4.list;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Objects;
+
+/**
+ * This class extends {@link TreeList}. It provides some methods that are needed
+ * in benchmarking the actual {@code TreeList} against the 
+ * {@link IndexedLinkedList}.
+ */
+public class ExtendedTreeList<E> extends TreeList<E> {
```

Comment:
Hello @coderodde,

This class doesn't belong in `main`. Why is this needed at all?

Reply:
Hello @garydgregory ,

I had to roll `ExtendedTreeList` in order to provide such methods as `getFirst()` and friends for the actual JMH benchmark. Shall I remove it from the PR?